### ISSUE TITLE
Automatically update this extension doesn't appear to work

### ISSUE THREAD

**Issue by jcansdale** [github/VisualStudio#877]
- Visual Studio 2017 RC4

1. Ensure that 'Automatically update this extension' is enabled:

![image](https://cloud.githubusercontent.com/assets/11719160/23189242/f486dac6-f889-11e6-912a-35ca33d09b86.png)

2. Check there is an update available:

![image](https://cloud.githubusercontent.com/assets/11719160/23189299/3766cb26-f88a-11e6-803b-7e2d95e88f81.png)

3. Restart Visual Studio

### Expected Behavior

The pending 'GitHub Extension for Visual Studio' would now be installed.

### Actual Behavior

The 'GitHub Extension for Visual Studio' still appears under 'Updates' and hasn't been automatically updated.

### Note

This extension has `AllUsers=true`, which I believe is a requirement when packging with the Visual Studio installer? When `AllUsers=false`, auto updates seem to work fine.

**Comment by jcansdale** [comment]
I've attempted to repro locally using the following steps.

1. Extract ZIP to `c:\AutoUpdateReproAllUsers`
[AutoUpdateReproAllUsers.zip](https://github.com/github/VisualStudio/files/791812/AutoUpdateReproAllUsers.zip)

2. Add new Extension Gallerie at `c:\AutoUpdateReproAllUsers\AllUsers.rss`
![image](https://cloud.githubusercontent.com/assets/11719160/23190204/72858068-f88e-11e6-98e0-1511f464bf66.png)

3. Install the 2.1.1.4 version of the extension:
![image](https://cloud.githubusercontent.com/assets/11719160/23190228/96cc4830-f88e-11e6-80f6-e974971f7af1.png)

4. Restart Visual Studio (extension installs with VSIXInstaller).

5. Extension is now visible in list with 'Automatically update this extension' checked.
![image](https://cloud.githubusercontent.com/assets/11719160/23190312/088c0302-f88f-11e6-938e-8af2e701c296.png)

6. Version `2.2.0.5` appears in list as available update.
![image](https://cloud.githubusercontent.com/assets/11719160/23190397/7fcf6d3c-f88f-11e6-952b-76ef6a3b7194.png)

7. Restart Visual Studio.

8. Version `2.2.0.5` remains in list of available updates and hasn't been automatically updated 😭.

Please let me know if we're doing wrong!

**Comment by tinaschrepfer** [comment]
Because VSIXes in VS 2017 are installed by the VS installer setup engine, you must have all instances of VS (all versions) closed for auto-update to work.  Auto-update kicks in every night and tries to update, but if you have VS open, then it will fail to do so.  We're working on improving the messaging of this for 15.1 so users will know that they have to shut down VS.

If you want to test this out yourself, you can shut down VS, go to Task Scheduler and manually run the VSIX Auto Update task manually.

**Comment by jcansdale** [comment]
Oh wow, thanks for the quick response!

Sure enough it's there:
![image](https://cloud.githubusercontent.com/assets/11719160/23191248/8fbdc144-f894-11e6-815d-71e356997056.png)

I never would have guessed. 😄

**Comment by jcansdale** [comment]
@tinaschrepfer, Great, thanks for letting us know!

Is seems automatic updates do work, I was just being impatient. 😉

**Comment by grokys** [comment]
@tinaschrepfer I installed VS2017 at launch last week and my version of GHfVS still hasn't been updated - it's still at 2.1.1.4:

![image](https://cloud.githubusercontent.com/assets/1775141/23866425/0f4f87ce-0819-11e7-87e3-dd6feb0ccd9c.png)

I enabled Task Scheduler history and I can see that the auto-update task is being run:

![image](https://cloud.githubusercontent.com/assets/1775141/23866466/37e9163c-0819-11e7-8340-52919b640e26.png)

Is there any more information I can provide to try and work out why this isn't updating?

**Comment by grokys** [comment]
I just noticed that the update task in the Task Scheduler is for VS2015, not 2017. That would explain it:

![image](https://cloud.githubusercontent.com/assets/1775141/24094110/7ca73e78-0d57-11e7-9695-5f5b053830c8.png)

Any idea how that happened, and how I can fix it?

**Comment by tinaschrepfer** [comment]
@grokys, are you saying that the 2017 task scheduler is not present?  When I go to VS in Task Scheduler, I see there's 2 versions, VSIX Auto Update 14 and VSIX Auto update 15.0.26220.0.  Is this not what you see?

**Comment by grokys** [comment]
@tinaschrepfer yep, exactly - when I go to VS in Task Scheduler, there's only the single entry as shown in the screenshot above. VS2015 and 2017 are installed on this machine.

**Comment by tinaschrepfer** [comment]
Ah, it looks like we only create the auto update task if you've installed any extension using VSIXInstaller.exe.  Prior to VS2017, GitHub was installed as part of the product using VSIXInstaller.exe.  But since then we've switched to directly use the VS installer setup engine.  I've created a bug to track this gap in behavior.  For now, you'll have to manually update GitHub extension initially, and then the auto update task will be created.  Also remember that for auto update to work, you'll need to shut down all instances of VS.  Sorry about this!
